drv/stm32h7-update-server/src/main.rs

```diff
@@ -0,0 +1,270 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// Functions for writing to flash for updates
+//
+// This driver is intended to carry as little state as possible. Most of the
+// heavy work and decision making should be handled in other tasks.
+#![no_std]
+#![no_main]
+
+use drv_update_api::UpdateError;
+use ringbuf::*;
+use stm32h7::stm32h753 as device;
+use userlib::*;
+
+const FLASH_KEY1: u32 = 0x4567_0123;
```

Comment:
I'd appreciate a comment here saying these specific values are specified / mandated by ST, not something that we picked ourselves, and citing section 4.5.1 in the reference manual.